testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterTest.java

```diff
@@ -491,6 +491,28 @@ public void registerUserNotUsernamePasswordPolicy() {
         assertEquals("Please specify username.", registerPage.getError());
     }
 
+    // KEYCLOAK-12729
+    @Test
+    public void registerUserNotEmailPasswordPolicy() {
+
+        RealmRepresentation notEmailRealm = RealmBuilder.create().passwordPolicy("notEmail").build();
+        notEmailRealm.setRegistrationEmailAsUsername(true);
```

Comment:
@thomasdarimont As you pointed, you are missing cleanup of those things after the test method.